### PR TITLE
Redesign settings layout

### LINKED ISSUE #507: Redesign Settings Layout
There are more and more options added to settings over time, making it a little bit lengthy. It may look snappy to show them on the sub-menus, only keeping the categories on the main menu.

### PR DESCRIPTION
**Pull request by morckx**
Fixes #507

Everything seems to work now, including search and back logic.

In a second PR we could re-organize a bit more, maybe.:

* add new sub-screen startup-up, including start-action and auto-play on start
* move MPD to connectivity
* move off-wifi-warning to connectivity

### CODE REVIEW

**Comment by werman** [comment]
Nice, I'll try to take a look later.

**Comment by morckx** [comment]
> Nice, I'll try to take a look later.

OK, take your time. It's not urgent, I guess, and apart from an automatic translation update (which makes more sense after this one), I haven't got anything in the pipeline for now.

**Comment by avently** [comment]
Looks cool! What purpose of the search field here?

**Comment by avently** [comment]
Oh, you added so many useless libs. Please, make this UI possible without external libs. Icons can be added one by one. Typeface can be changed to one of the stock typefaces without visual impact. SearchPreference is useless. These libs just wasting APK size.

**Comment by morckx** [comment]
> Oh, you added so many useless libs. 

Which one is useless?

> Please, make this UI possible without external libs. Icons can be added one by one. 

Sorry, but that would be too much work for work for me, but feel free to provide something more size efficient. 

Iconics, however, opens the way for very flexible and easy icon handling for current (new player screen) and future additions and is API<21-compatible. A compromise might be to use an older version of Iconics.
 
>Typeface can be changed to one of the stock typefaces without visual impact.

The PR uses Iconics only for icons.

> SearchPreference is useless. These libs just wasting APK size.

How much space does it take? I don't think it's very much.

**Comment by morckx** [comment]
>> SearchPreference is useless. These libs just wasting APK size.

> How much space does it take? I don't think it's very much.

APK size is only increased by 250kb. Concerning the memory consumption, I couldn't see any difference.

File: app/src/main/java/net/programmierecke/radiodroid2/FragmentSettings.java [hunk 1]
```diff
@@ -36,83 +45,155 @@
...
-                ((ActivityMain)getActivity()).getToolbar().setTitle(R.string.settings_statistics);
-                FragmentServerInfo f = new FragmentServerInfo();
-                FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
-                fragmentTransaction.replace(R.id.containerView, f).addToBackStack(String.valueOf(FRAGMENT_FROM_BACKSTACK)).commit();
-                return false;
-            }
-        });
+    private void refreshToplevelIcons() {
+        findPreference("shareapp_package").setSummary(getPreferenceManager().getSharedPreferences().getString("shareapp_package", ""));
+        findPreference("pref_category_ui").setIcon(Utils.IconicsIcon(getActivity(), "cmd_monitor"));
```
Comment (werman) [inline app/src/main/java/net/programmierecke/radiodroid2/FragmentSettings.java]: Better use enum like `CommunityMaterial.Icon2.cmd_monitor`and not hardcoded strings.

File: app/build.gradle [hunk 2]
```diff
@@ -23,6 +23,7 @@ android {
         versionCode 85
         versionName "0.75"
 
+        multiDexEnabled true
```
Comment (werman) [inline app/build.gradle]: Huh, to many classes?

Comment (morckx) [reply app/build.gradle]: Yes, the many classes come the new Kotlin based Iconics version v4.0. I could just revert to v3.2.5, but sooner or later, like most other projects, we will have to bite the bullet anyway, and I don't think it's that bad anymore.

What do you think?

File: app/src/main/java/net/programmierecke/radiodroid2/FragmentSettings.java (hunk 1 above)
Comment (morckx) [reply app/src/main/java/net/programmierecke/radiodroid2/FragmentSettings.java]: indeed, will change...

**Comment by morckx** [comment]
> On github of that search library I see description strings in search, could we do this?

Right, I'll have a look...

File: app/build.gradle (hunk 2 above)
Comment (werman) [reply app/build.gradle]: It doesn't really harm anything so I don't care much.

**Comment by morckx** [comment]
> > On github of that search library I see description strings in search, could we do this?
> 
> Right, I'll have a look...

It shows the summaries, but all our summaries include the settings value so I deliberately set `search:summary=""` to avoid that sth like "%s" is shown.

There is also an option to add breadcrumbs, but with our tree depth of two, it doesn't make sense, I guess.

**Pull request merged**